### DISCUSSION TITLE
Use uvicorn or gunicorn in docker CMD

### DISCUSSION THREAD

**Discussion by luozhouyang** [fastapi/fastapi#7527] [Questions] [answered]
### First check

* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description
I read the document [Deployment](https://fastapi.tiangolo.com/deployment/)

<img width="770" alt="截屏2020-04-14 下午4 55 27" src="https://user-images.githubusercontent.com/34032031/79205644-cb45af00-7e70-11ea-9345-075a9ea181f8.png">

The CMD runs in docker use `uvicorn`，why not `gunicorn`？
Which should i use for best performance?

**Comment by phy25** [comment]
#936: gunicorn is a "Python WSGI HTTP Server for UNIX" while fastapi uses ASGI.

**Comment by luozhouyang** [comment]
But why the base image `tiangolo/uvicorn-gunicorn-fastapi:python3.7` needs `gunicorn`?

**Comment by Zheaoli** [comment, marked as answer]
@luozhouyang this just let people to use both uvicorn  and gunicorn  bu using the same image to adapt different circumstance

**Comment by luozhouyang** [comment]
I see. Thanks!

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :clap: :bow: 

Thanks for reporting back and closing the issue @luozhouyang  :+1: 

About the Docker image, in short, Gunicorn starts several Uvicorn workers, it makes sure there are several processes ready to work. And each of those Uvicorn workers do the actual work.

You can read more in the docs for the Docker image: https://github.com/tiangolo/uvicorn-gunicorn-fastapi-docker
